pkg/arduino_adafruit_sensor: add Adafruit Unified Sensor Driver as package

gschorcht opened the pull request:
### Contribution description

This PR provides the [Adafruit Unified Sensor Driver](https://github.com/adafruit/Adafruit_Sensor) as package.

There are a number of Adafruit sensor drivers which all use a common base class `Adafruit_Sensor` from the [Adafruit Unified Sensor Driver](https://github.com/adafruit/Adafruit_Sensor). To support such drivers, the Adafruit Unified Sensor Driver is provided as package.

Adafruit sensor driver for ST LSM9DS0 will be provided as separat PR as package for demonstration and testing.

PR #12518 will be rebased for testing to have an Adafruit sensor driver for a sensor for which there is a native driver in RIOT.

### Testing procedure

Use a board that provides the `arduino` feature and flash
```
BOARD=... make -C tests/pkg_arduino_adafruit_sensor flash test
```
PR #12518 can be used as a more complex test for using the package.

### Issues/PRs references

Prerequisite for PR #12518

fjmolinas commented on pkg/arduino_adafruit_sensor/Makefile.include:
```diff
@@ -0,0 +1,2 @@
+CFLAGS += "-DARDUINO = 100"
```
What does this do?

gschorcht replied:
It just says the source code of the package that we have an Arduino version > 1.0. This is usually set by the Arduino IDE.

fjmolinas replied:
Mind adding a comment? Squashing out that `f` and triggering ci?

fjmolinas reviewed (commented):
Looks good, one tiny questio above, also I see a trailing f in the commit message of fa7d4d9a067d0c65327b7f02d52e725c05905ce3 that bothers my TOC

gschorcht commented:
> Looks good, one tiny questio above, also I see a trailing f in the commit message of [fa7d4d9](https://github.com/RIOT-OS/RIOT/commit/fa7d4d9a067d0c65327b7f02d52e725c05905ce3) that bothers my TOC

Ok, after almost a year I rebased the PR onto the latest master and fixed the comment in the commit messages. The only change is that I upgraded the package to the latest version of the package.

benpicco commented:
bors merge

gschorcht commented:
Thanks for reviewing and merging.
